packages/node_modules/@node-red/editor-client/src/js/ui/view.js

```diff
@@ -3505,491 +3505,464 @@ RED.view = (function() {
-                        node[0][0].__iconShade__ = icon_shade;
-
-                        var icon_group = d3.select(icon_groupEl)
-                        createIconAttributes(icon_url, icon_group, d);
-
-                        var icon_shade_border = document.createElementNS("http://www.w3.org/2000/svg","path");
-                        icon_shade_border.setAttribute("d","right" != d._def.align ? "M 30 1 l 0 "+(d.h-2) : "M 0 1 l 0 "+(d.h-2)  )
-                        icon_shade_border.setAttribute("class", "red-ui-flow-node-icon-shade-border")
-                        icon_groupEl.appendChild(icon_shade_border);
-                        node[0][0].__iconShadeBorder__ = icon_shade_border;
-
-                        nodeContents.appendChild(icon_groupEl);
-                    }
-                    var labelLineNumber = (separateTextByLineBreak.length == 0)? 1:separateTextByLineBreak.length;
-                    var labelId = d.id.replace(".","-");
-                    for(var i=0;i<labelLineNumber;i++) {
-                        var text = document.createElementNS("http://www.w3.org/2000/svg","text");
-                        text.setAttribute("class","red-ui-flow-node-label"+(hideLabel?" hide":"")+(d._def.align?" red-ui-flow-node-label-"+d._def.align:""));
-                        text.setAttribute("id", "red-ui-flow-node-label-"+labelId+"-"+i);
-                        text.setAttribute("x", 38);
-                        text.setAttribute("dy", ".3px");
-                        text.setAttribute("text-anchor",d._def.align !== "right" ? "start":"end");
-                        nodeContents.appendChild(text);
-                    }
+                    var thisNode = d3.select(this);
+                    //thisNode.selectAll(".centerDot").attr({"cx":function(d) { return d.w/2;},"cy":function(d){return d.h/2}});
+                    this.setAttribute("transform", "translate(" + (d.x-d.w/2) + "," + (d.y-d.h/2) + ")");
+                    if (mouse_mode != RED.state.MOVING_ACTIVE) {
+                        this.classList.toggle("red-ui-flow-node-disabled", d.d === true);
+                        this.classList.toggle("red-ui-flow-subflow", activeSubflow != null)
+                        this.classList.toggle("red-ui-flow-node-selected", !!d.selected )
+                        this.__mainRect__.setAttribute("width", d.w)
+                        this.__mainRect__.setAttribute("height", d.h)
+                        this.__mainRect__.classList.toggle("red-ui-flow-node-highlighted",!!d.highlighted );
+
+                        var text = thisNode.selectAll(".red-ui-flow-node-label");
+                        thisNode.selectAll(".red-ui-flow-node-label-text").remove();
+                        var sa = convertLineBreakCharacter(label);
+                        var sn = sa.length;
+                        var yp = d.h / 2 - (sn / 2) * 24 + 16
+                        sa.forEach(function (line) {
+                            text.append("tspan")
```

Comment:
I just did a big refactor of this file to remove as much `d3` as possible. There's no reason to add more back. This should be a `document.createElementNS(...)` call and then set the attributes.